main/src/main/scala/sbt/Main.scala

```diff
@@ -880,10 +932,14 @@ object BuiltinCommands {
     val session = Load.initialSession(structure, eval, s0)
     SessionSettings.checkSession(session, s2)
     val s3 = addCacheStoreFactoryFactory(Project.setProject(session, structure, s2))
-    val s4 = LintUnused.lintUnusedFunc(s3)
-    CheckBuildSources.init(s4)
+    val s4 = setupGlobalFileTreeRepository(s3)
```

Comment:
@eatkins This line, which closes the current GlobalFileTreeRepository and creates a new one **after** `onLoad` is executed, is causing problems. Please see #6190, thanks!